### ISSUE TITLE
Loss of network connectivity between pods with ipsec enabled and ip xfrm state anti-replay oseq maxed out at 0xffffffff

### ISSUE THREAD

**Issue by henrysfx** [cilium/cilium#12447]
## Bug report

**General Information**

- Cilium version (run `cilium version`): 1.7.5
- Kernel version (run `uname -a`): 4.14.127+
- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...): Server v1.14.10-gke.36

**Relevant output from `ip -s x s`**

_BEFORE** (connection refused is expected, just to show bi-directional connectivity):_
```
Tue Jul 7 18:26:16 UTC 2020
src 10.11.13.199 dst 10.11.8.249
	proto esp spi 0x00000001(1) reqid 1(0x00000001) mode tunnel
	replay-window 0 seq 0x00000000 flag (0x00000000)
	mark 0x1e00/0xff00
	aead rfc4106(gcm(aes)) 0xXXXX (160 bits) 128
	anti-replay context: seq 0x0, oseq 0xffffbd4f, bitmap 0x00000000
	sel src 0.0.0.0/0 dst 0.0.0.0/0 uid 0
	lifetime config:
	 limit: soft (INF)(bytes), hard (INF)(bytes)
	 limit: soft (INF)(packets), hard (INF)(packets)
	 expire add: soft 0(sec), hard 0(sec)
	 expire use: soft 0(sec), hard 0(sec)
	lifetime current:
	 1783270055274(bytes), 4294950222(packets)
	 add 2020-06-27 04:04:25 use 2020-06-27 04:04:25
	stats:
	 replay-window 0 replay 0 failed 0
src 10.11.13.199 dst 10.11.68.95
10.11.8.203: inverse host lookup failed: 
(UNKNOWN) [10.11.8.203] 9095 (?) : Connection refused
```

_AFTER (now hangs, ESP response packet never hits the wire):_
```
Tue Jul 7 18:26:17 UTC 2020
src 10.11.13.199 dst 10.11.8.249
	proto esp spi 0x00000001(1) reqid 1(0x00000001) mode tunnel
	replay-window 0 seq 0x00000000 flag (0x00000000)
	mark 0x1e00/0xff00
	aead rfc4106(gcm(aes)) 0xXXXX (160 bits) 128
	anti-replay context: seq 0x0, oseq 0xffffffff, bitmap 0x00000000
	sel src 0.0.0.0/0 dst 0.0.0.0/0 uid 0
	lifetime config:
	 limit: soft (INF)(bytes), hard (INF)(bytes)
	 limit: soft (INF)(packets), hard (INF)(packets)
	 expire add: soft 0(sec), hard 0(sec)
	 expire use: soft 0(sec), hard 0(sec)
	lifetime current:
	 1783293126142(bytes), 4294967295(packets)
	 add 2020-06-27 04:04:25 use 2020-06-27 04:04:25
	stats:
	 replay-window 0 replay 0 failed 0
src 10.11.13.199 dst 10.11.68.95
10.11.8.203: inverse host lookup failed: 
(UNKNOWN) [10.11.8.203] 9095 (?) : Connection timed out
```

**How to reproduce the issue**

1. install cilium with helm chart @ 1.7.5 with following options
```
export NATIVE_CIDR=$(gcloud container clusters describe $CLUSTER_NAME --region $CLUSTER_REGION | grep -i clusterIpv4Cidr | awk '{print $2}' | head -n 1)
kubectl create -n cilium secret generic cilium-ipsec-keys \
    --from-literal=keys="1 rfc4106(gcm(aes)) $(echo $(dd if=/dev/urandom count=20 bs=1 2> /dev/null| xxd -p -c 64)) 128"
helm install cilium cilium/cilium --version 1.7.5 \
    --debug \
    --namespace cilium \
    --set global.etcd.enabled=true \
    --set global.etcd.managed=true \
    --set global.identityAllocationMode=kvstore \
    --set global.kubeProxyReplacement=disabled \
    --set global.ipv6.enabled=false \
    --set nodeinit.reconfigureKubelet=true \
    --set nodeinit.removeCbrBridge=false \
    --set nodeinit.revertReconfigureKubelet=true \
    --set global.cni.binPath=/home/kubernetes/bin \
    --set global.nativeRoutingCIDR=$NATIVE_CIDR \
    --set global.psp.enabled=true \
    --set global.prometheus.enabled=true \
    --set global.operatorPrometheus.enabled=true \
    --set global.encryption.enabled=true \
    --set global.encryption.interface=eth0 \
    --set global.encryption.nodeEncryption=false \
    --set global.encryption.secretName=cilium-ipsec-keys \
    --set global.gke.enabled=true \
    --set global.nodeinit.enabled=true \
    --set global.nodePort.mode=snat \
    --set global.debug.enabled=false \
    --set global.tunnel=disabled
```
2. generate load between any 2 cilium-managed pods
3. monitor ipsec tunnel oseq number until it reaches `0xffffffff`
`ip -s x s | grep -A 16 "src <source node cilium_host IP> dst <dest node cilium_host IP>" | grep oseq`

At this point, connectivity is lost as ESP packets in the direction which the tunnel has reached `0xffffffff` is dropped and no longer sent at all, resulting in network connectivity loss/timeout.

**Comment by errordeveloper** [comment]
@henrysfx have you tried to reproduce it with 1.8.x, or perhaps 1.9.0?

**Comment by tgraf** [comment]
This has likely been fixed by https://github.com/cilium/cilium/pull/15039

**Comment by pchaigno** [comment]
The fix will be in the next v1.8.8 and v1.9.5. You can also already test with `:latest`.
